src/main/java/pl/cleankod/exchange/core/domain/Currency.java

```diff
@@ -0,0 +1,21 @@
+package pl.cleankod.exchange.core.domain;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonValue;
+
+public enum Currency {
+
+    PLN, EUR, THB, USD, AUD, HKD, CAD, NZD, SGD, HUF, GBP, UAH, JPY, CZK, DKK, ISK, NOK, SEK, RON, BGN, TRY, ILS, CLP, PHP, MXN, ZAR, BRL, MYR, IDR, INR, KRW, CNY, XDR;
+
+    @JsonCreator
+    public static Currency fromString(String currency) {
+        return valueOf(currency.toUpperCase());
```

Comment:
Then it would be swalloed by ControllerAdvice, and provded some error code to frontend. 
wth java.util.Currency I had problems that openapi was exposing irrelevant fields, so I decieded to change it to not waste more time on incorrect contract exposure